components/script/textinput.rs

```diff
@@ -850,8 +850,12 @@ impl<T: ClipboardProvider> TextInput<T> {
         } else {
             vec!(content)
         };
-        self.edit_point.line = min(self.edit_point.line, self.lines.len() - 1);
-        self.edit_point.index = min(self.edit_point.index, self.current_line_length());
+        if update_text_cursor {
+            self.edit_point.line = min(self.edit_point.line, self.lines.len() - 1);
+            self.edit_point.index = min(self.edit_point.index, self.current_line_length());
+        } else {
+            self.edit_point = Default::default();
```

Comment:
I don't think we want to reset this to 0 - instead, we should just not update the edit point.

Reply:
But not resetting this to default is giving wrong output. And i am not able to know how does edit_point's value gets updated before it enters the children_changed function. This should not happen since children_changed calls reset which calls update_text_contents which finally calls setValue.
But explicitly setting this to default gives correct output for all tests including wpt tests.